### PR TITLE
REFACTOR: reformat into python module, clean up visualization

### PR DESCRIPTION
**Pull request by madisonmay**

### CODE REVIEW

File: enso/experiment/__init__.py [hunk 1]
```diff
@@ -80,9 +83,12 @@ def _dump_results(self):
         result_file = "%s/Results.csv" % result_path
         self.results.to_csv(result_file)
 
-        # The a is for archival, not just a typo
+        # The a is for archival, not just a typoOI
```
Comment (Slater-Victoroff) [inline enso/experiment/__init__.py]: typoOI

File: enso/featurize/transformer_features.py [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+# package when implementation / saved weights are stable
+import_path = os.path.abspath(
+    os.path.join(
+        os.path.dirname(os.path.dirname(transformer.__file__)),
+        "data/models/SNLI"
+    )
+)
+sys.path.append(import_path)
+
+from v1.model import Transformer
```
Comment (Slater-Victoroff) [inline enso/featurize/transformer_features.py]: Should probably add this to Enso. Maybe we have a directory like the ML directory in indicoAPI that is for source files like these. Would be very surprised if this is the last time this happens.

Potentially we also make a privvy entry for the weight file for maximum reproducability.

File: enso/featurize/transformer_features.py [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+
+    initialized = False
+
+    def __init__(self, *args, **kwargs):
+        super(Featurizer).__init__(*args, **kwargs)
+        TransformerFeaturizer.setup()
+
+    @classmethod
+    def setup(cls):
+        print("Setting up transformer Featurizer...")
```
Comment (Slater-Victoroff) [inline enso/featurize/transformer_features.py]: Anything preventing us from handling this in a logger?

File: enso/utils.py [hunk 4]
```diff
@@ -18,15 +20,22 @@ def feature_set_location(dataset_name, featurizer):
...
+
     relevant_classes = []
     for filename in files:
-        if "__init__" not in filename and not filename.endswith('.pyc'):
-            module_name = filename.rpartition('.')[0]
-            mod = import_module("%s.%s" % (root, module_name))
-            for name, obj in inspect.getmembers(mod):
-                if inspect.isclass(obj) and obj.__name__ in match_names:
-                    relevant_classes.append(obj)
+        if filename.endswith('.pyc'):
```
Comment (Slater-Victoroff) [inline enso/utils.py]: We should probably shift this logic to be affirmative rather than exclusionary.

File: enso/experiment/__init__.py (hunk 1 above)
Comment (madisonmay) [reply enso/experiment/__init__.py]: This is great.

File: enso/featurize/transformer_features.py (hunk 3 above)
Comment (madisonmay) [reply enso/featurize/transformer_features.py]: we don't need those tbh -- just removed them

File: enso/featurize/transformer_features.py (hunk 2 above)
Comment (madisonmay) [reply enso/featurize/transformer_features.py]: I'm not sure what the right approach is here.  Adding all the experimental models into enso directly seems... undesirable.  Among other things it locks us into specific library versions between featurizer projects.

Comment (madisonmay) [reply enso/featurize/transformer_features.py]: I can remove this file from the PR if you like.

File: enso/utils.py (hunk 4 above)
Comment (madisonmay) [reply enso/utils.py]: :+1:

Comment (madisonmay) [reply enso/utils.py]: Revised implementation could be:

```
        if not filename.endswith('.py') or '__' in filename:
            continue
```

File: enso/featurize/transformer_features.py (hunk 2 above)
Comment (Slater-Victoroff) [reply enso/featurize/transformer_features.py]: hmmm. Yea, that's a good point. Not really sure what the best way of handling this is. Once it's in indico API it's easy enough. I think I'd rather include experiments than remove things alltogether, but at the same time it should be a pretty limited to make it not awful

**Pull request merged**